RedHatInsights/patchman-ui
# chore: use paginated requests to fetch all template names

# Description

Associated Jira ticket: # (issue)

Fetches all template names using paginated requests


# How to test the PR

1. Visit the templates page
2. Try creating a new template
3. Observe that 
     - the next button is disabled unless all taken/reserved template names are fetched from the API 
     - spinner icon is displayed while the requests are made
     - once all requests are resolved, the name input is validated
4. Visit some template detail page
5. Try adding more systems to an existing template with systems assigned
6. Observe that the already assigned systems are marked as selected

# Before the change


# After the change


# Dependent work link


# Checklist:
